### PR TITLE
add entitiesAt() for scatter plot

### PR DESCRIPTION
**Pull request by ztsai**
closes: #2569
demo: http://jsfiddle.net/fa3mjzew/4/

### CODE REVIEW

File: test/plots/scatterPlotTests.ts [hunk 1]
```diff
@@ -214,6 +214,71 @@ describe("Plots", () => {
...
+
+      svg.remove();
+    });
+
+    it("determines whether an entity contains a given point by its size", () => {
+      let svg = TestMethods.generateSVG(400, 400);
+      let xScale = new Plottable.Scales.Linear();
+      let yScale = new Plottable.Scales.Linear();
+
+      let dataset = new Plottable.Dataset([{x: 0, y: 0}, {x: 1, y: 1}]);
```
Comment (jtlan) [inline test/plots/scatterPlotTests.ts]: Is this `Dataset` needed for this test? It looks like the later `entitiesAt()` calls are only testing against a point in the other `Dataset`.

File: test/plots/scatterPlotTests.ts [hunk 2]
```diff
@@ -214,6 +214,71 @@ describe("Plots", () => {
...
+      assert.lengthOf(entities, 1, "only one entity has been retrieved");
+      assert.deepEqual(entities[0].datum, {x: 1, y: 2}, "correct datum has been retrieved");
+
+      plot.size(6);
+      entities = plot.entitiesAt({ x: xScale.scale(1) + 5, y: yScale.scale(2) - 5});
+      assert.lengthOf(entities, 0, "none of the entities is retrieved");
+      svg.remove();
+    });
+
+    it("returns an empty array when no entities contain a given point", () => {
```
Comment (jtlan) [inline test/plots/scatterPlotTests.ts]: Isn't this basically the same thing as the last `assert()` in the above test?

File: test/plots/scatterPlotTests.ts [hunk 3]
```diff
@@ -214,6 +214,71 @@ describe("Plots", () => {
       svg.remove();
     });
 
+    it("can retrieve entities centered at a given point", () => {
+      let svg = TestMethods.generateSVG(400, 400);
+      let xScale = new Plottable.Scales.Linear();
+      let yScale = new Plottable.Scales.Linear();
+
+      let dataset = new Plottable.Dataset([{x: 0, y: 0}, {x: 1, y: 1}]);
+      let dataset2 = new Plottable.Dataset([{x: 1, y: 2}, {x: 3, y: 4}]);
```
Comment (jtlan) [inline test/plots/scatterPlotTests.ts]: This `Dataset` doesn't appear to be used in the test. Can we remove it?

**Comment by jtlan** [comment]
In some of the tests, multiple `Dataset`s are being used, but the scope of the test only requires one to be present, meaning the test setup can probably be streamlined a bit.

However, we should also have a test that confirms that `entitiesAt()` correctly returns overlapping points from multiple `Dataset`s.

File: src/plots/scatterPlot.ts [hunk 4]
```diff
@@ -180,6 +180,27 @@ export module Plots {
...
+      let yProjector = Plot._scaledAccessor(this.y());
+      let sizeProjector = Plot._scaledAccessor(this.size());
+      return this.entities().filter((entity) => {
+        let datum = entity.datum;
+        let index = entity.index;
+        let dataset = entity.dataset;
+        let x = xProjector(datum, index, dataset);
+        let y = yProjector(datum, index, dataset);
+        let size = sizeProjector(datum, index, dataset);
+        return x - size / 2  <= p.x && p.x <= x + size / 2 && y - size / 2 <= p.y && p.y <= y + size / 2;
```
Comment (jtlan) [inline src/plots/scatterPlot.ts]: I made [a comment](https://github.com/palantir/plottable/issues/2569#issuecomment-130851513) on the root issue about whether we should be treating the points as "square" (the logic here) or "round". That would require a change to this line and probably also the test code.

**Comment by crmorford** [comment]
That's an amazing demo fiddle!
I was going to test different shapes, sizes, and overlapping symbols, and you hit them all with one test. 
Super helpful, thank you :)

**Comment by crmorford** [comment]
tested on all browsers, with multiple datasets, etc. looks good :)

**Pull request merged**